draft/XXXX-missing-values-in-typed-cols/README.md

```diff
@@ -0,0 +1,44 @@
+---
+RFC: XXXX
+Author: Jack Wilburn
+Created: 2020-10-26
+Last-Modified: 2020-10-26
+---
+
+# RFC XXXX: Missing Values In Typed Columns
+
+As we move to having typed data in Multinet, it is important to handle missing data elegantly, even with typing. Currently, if a column has any amount of missing data, we assume that the data is a "label" type. Since the "label" type is essentially a catch-all for data that we can't provide a meaningful type for, it's important that if there is a semantically meaningful type, we provide it to visualization applications. This must, also, be true in the case where we have mostly complete data with just a few missing rows. Without this information, the client applications cannot adequately choose a visualization for these columns.
+
+There will likely be some client- and server-side considerations for handling missing data. This will include correctly parsing data and when reading it into the ArangoDB types (or we may choose to leave them as strings in the db). Types that we define, will be referenced as semantic types, and native ArangoDB types will be referenced as Arango types
+
+## Motivation
+
+We now have a motivation to supply data schema information to visualization applications, UpSet. Upset requires set membership information and other data types to generate its visualization. We can also utilize this type information in the other visualization applications that support multinet (MultiLink and MultiMatrix).
+
+Since we now need to pass semantic typing into these applications, knowing what the types are, even if there is missing data, is essential. As such, we'll need to modify our typing logic to handle missing data. This stretches from the frontend, where we provide type suggestions, to the backend, where we store the type information and disseminate it to the vis applications.
```

Comment:
I don't think this is the right motivation. It's more general: we want to support typed columns, and in the real world datasets almost always have missing entries, so we'll need to support that reality. This has very little to do with UpSet or any other specific vis application.

Reply:
Updated in 2872e82

Reply:
Just FYI, that commit does not seem to address my original comment.

Reply:
Ah I think you meant 535593f6a7ca1666dd52d51928dfe0d130413c36. Looks good, thanks.

Reply:
Whoops, yes